src/main/java/com/thevoxelbox/voxelsniper/SnipeData.java

```diff
@@ -24,41 +24,55 @@
  */
 package com.thevoxelbox.voxelsniper;
 
+import com.thevoxelbox.voxelsniper.util.BlockHelper;
 import com.thevoxelbox.voxelsniper.util.VoxelList;
 
 import org.spongepowered.api.Sponge;
 import org.spongepowered.api.block.BlockState;
 import org.spongepowered.api.block.BlockTypes;
-import org.spongepowered.api.data.key.Key;
-import org.spongepowered.api.data.value.BaseValue;
+import org.spongepowered.api.block.trait.BlockTrait;
 import org.spongepowered.api.text.Text;
 import org.spongepowered.api.world.World;
 
-import java.util.Optional;
+import java.util.*;
 
 public class SnipeData {
 
     private Sniper owner;
     private Message voxelMessage;
 
-    private double brushSize = VoxelSniperConfiguration.DEFAULT_BRUSH_SIZE;
-    private BlockState voxelId = Sponge.getRegistry().getType(BlockState.class, VoxelSniperConfiguration.DEFAULT_VOXEL_ID).orElse(BlockTypes.AIR.getDefaultState());
-    private BlockState replaceId =
-            Sponge.getRegistry().getType(BlockState.class, VoxelSniperConfiguration.DEFAULT_REPLACE_ID).orElse(BlockTypes.AIR.getDefaultState());
-    private VoxelList voxelList = new VoxelList();
-    private Key<?> voxelInkKey = null;
-    private Object voxelInkValue = null;
-    private Key<?> replaceInkKey = null;
-    private Object replaceInkValue = null;
-
-    private int voxelHeight = VoxelSniperConfiguration.DEFAULT_VOXEL_HEIGHT;
-    private int cCen = VoxelSniperConfiguration.DEFAULT_CYLINDER_CENTER;
-    private int range = 0;
-    private boolean ranged = false;
-    private boolean lightning = false;
+    private double brushSize;
+    private BlockState voxelState;
+    private BlockState replaceState;
+    private VoxelList voxelList;
+    private Map<BlockTrait<?>, Object> voxelInkTraits;
+    private Map<BlockTrait<?>, Object> replaceInkTraits;
+
+    private int voxelHeight;
+    private int cCen;
+    private int range;
+    private boolean ranged;
+    private boolean lightning;
 
     public SnipeData(Sniper vs) {
-        this.owner = vs;
+        owner = vs;
+
+        voxelState = Sponge.getRegistry().getType(BlockState.class, VoxelSniperConfiguration.DEFAULT_VOXEL_ID)

```

Comment:
I get that a majority of this is moving the defaults being set on field initializers versus the constructor, but it's cluttering the actual changes you're making, which is the ink trait maps. For the sake of reviewing the PR, I'd like to keep the changes to a minimum, and this sort of change is extra noise.

Reply:
This change has been split to a new PR.